buttercup/cogs/find.py

```diff
@@ -241,18 +232,87 @@ async def _find(self, ctx: SlashContext, reddit_url: str) -> None:
             )
             return
 
-        await msg.edit(content="I found the post!", embed=self.to_embed(data))
+        await msg.edit(content="I found the post!", embed=to_embed(data))
 
         # If we started with a transcription link, this will be set already.
         if "transcription" not in data:
             data: Dict[str, Dict] = self.get_transcription(data)
 
-        await msg.edit(content="I found the post!", embed=self.to_embed(data))
+        await msg.edit(content="I found the post!", embed=to_embed(data))
 
         if data.get("transcription") or data["submission"].get("claimed_by"):
             data: Dict[str, Dict] = self.get_author(data)
 
-        await msg.edit(content="I found the post!", embed=self.to_embed(data))
+        await msg.edit(content="I found the post!", embed=to_embed(data))
+
+    @cog_ext.cog_slash(
+        name="claimed",
+        description="Find the posts claimed by the given user.",
+        options=[
+            create_option(
+                name="username",
+                description="The user to find the claimed posts for.",
+                option_type=3,
+                required=True,
+            )
+        ],
+    )
+    async def _claimed(self, ctx: SlashContext, username: str) -> None:
+        # Send a first message to show that the bot is responsive.
+        # We will edit this message later with the actual content.
+        msg = await ctx.send(
+            i18n["claimed"]["looking_for_claimed_posts"].format(username)
+        )
+
+        volunteer_response = self.blossom_api.get_user(username)
+        if volunteer_response.status != BlossomStatus.ok:
+            await msg.edit(content=i18n["claimed"]["user_not_found"].format(username))
+            return
+        volunteer = volunteer_response.data
+
+        # We can't specify that no user completed the post yet.
+        # Instead, we ask that the post isn't archived yet.
+        # For a short amount of time this might return completed posts.
+        submission_response = self.blossom_api.get(
+            "submission/", params={"claimed_by": volunteer["id"], "archived": False}
```

Comment:
Wouldn't it be better to update Blossom with having `completed_by` as a parameter? This method would be reduced in size quite drastically.

Reply:
It does have `completed_by`, but I can't filter for the value to be `NULL`, at least AFAIK

Reply:
That does make sense... We could make a calculated boolean property which resembles the current state of the post? e.g. unclaimed, in progress, and done, such as the subreddit